src/events.ts

```diff
@@ -73,9 +73,10 @@ async function processEvents() {
 
   for (const event of events) {
     const proposalId = event.id.replace('proposal/', '');
-    const proposal = await getProposal(proposalId);
+    let proposal = await getProposal(proposalId);
     const subscribers = await getSubscribers(event.space);
-
+    if (event.event === 'proposal/deleted')
+      proposal = { id: proposalId, space: { id: event.space } };
```

Comment:
This could be improved with

```diff
diff --git a/src/events.ts b/src/events.ts
index 4de90e2..2a30e08 100644
--- a/src/events.ts
+++ b/src/events.ts
@@ -73,11 +73,16 @@ async function processEvents() {
 
   for (const event of events) {
     const proposalId = event.id.replace('proposal/', '');
-    let proposal = await getProposal(proposalId);
-    const subscribers = await getSubscribers(event.space);
-    if (event.event === 'proposal/deleted')
+    let proposal: Record<string, any> | null;
+
+    if (event.event === 'proposal/deleted') {
       proposal = { id: proposalId, space: { id: event.space } };
+    } else {
+      proposal = await getProposal(proposalId);
+    }
+
     if (proposal) {
+      const subscribers = await getSubscribers(event.space);
       providers.forEach(provider => {
         provider(event, proposal, subscribers);
       });
```

We will check save a call to `getProposal` on `proposal/deleted` events, and save a call to `getSubscribers` when proposal is not found